app/modules/sale.js

```diff
@@ -1,57 +1,120 @@
+  showErrorNotification,
+  showInfoNotification,
+  showSuccessNotification
+} from './notifications'
 import { getWIF, LOGOUT } from './account'
 import { getNetwork } from './metadata'
-import { getNEO } from './wallet'
+import { getNEO, getGAS } from './wallet'
+import { toNumber, toBigNumber } from '../core/math'
 import asyncWrap from '../core/asyncHelper'
 
-// TODO: Rewrite this function
-
-export const participateInSale = (neoToSend: number, scriptHash: string) => async (dispatch: DispatchType, getState: GetStateType) => {
+export const participateInSale = (
+  neoToSend: number,
+  gasToSend: number,
+  scriptHash: string
+) => async (dispatch: DispatchType, getState: GetStateType) => {
   const state = getState()
   const wif = getWIF(state)
   const NEO = getNEO(state)
+  const GAS = getGAS(state)
   const net = getNetwork(state)
 
   const account = new wallet.Account(wif)
-  if (parseFloat(neoToSend) !== parseInt(neoToSend)) {
-    dispatch(showErrorNotification({ message: 'You cannot send fractional NEO to a token sale.' }))
+  if (neoToSend && parseFloat(neoToSend) !== parseInt(neoToSend)) {
+    dispatch(
+      showErrorNotification({
+        message: 'You cannot send fractional NEO to a token sale.'
+      })
+    )
+    return false
+  }
+
+  const neoToMint = toNumber(neoToSend)
+  const gasToMint = toNumber(gasToSend)
+
+  if ((neoToMint && isNan(neoToMint)) || (gasToMint && isNan(gasToMint))) {
+    dispatch(
+      showErrorNotification({ message: 'Please enter valid numbers only' })
+    )
+    return false
+  }
+
+  if (neoToMint > NEO) {
+    dispatch(
+      showErrorNotification({ message: 'You do not have enough NEO to send.' })
+    )
     return false
   }
-  const toMint = parseInt(neoToSend)
-  // $FlowFixMe
-  if (toMint > NEO) {
-    dispatch(showErrorNotification({ message: 'You do not have enough NEO to send.' }))
+
+  if (gasToMint > GAS) {
+    dispatch(
+      showErrorNotification({ message: 'You do not have enough GAS to send.' })
+    )
     return false
   }
+
   if (scriptHash.slice(0, 1) !== '0x' && scriptHash.length !== 42) {
     dispatch(showErrorNotification({ message: 'Not a valid script hash.' }))
     return false
   }
   const _scriptHash = scriptHash.slice(2, scriptHash.length)
 
-  dispatch(showInfoNotification({ message: 'Sending transaction', autoDismiss: 0 }))
+  dispatch(
+    showInfoNotification({ message: 'Sending transaction', autoDismiss: 0 })
+  )
 
   const [error, rpcEndpoint] = await asyncWrap(api.neonDB.getRPCEndpoint(net)) // eslint-disable-line
-  const [err, balance] = await asyncWrap(api.nep5.getTokenBalance(rpcEndpoint, _scriptHash, account.address)) // eslint-disable-line
-  const [e, response] = await asyncWrap(api.neonDB.doMintTokens(net, _scriptHash, wif, toMint, 0))
+  const [err, balance] = await asyncWrap(
+    api.nep5.getTokenBalance(rpcEndpoint, _scriptHash, account.address)
+  ) // eslint-disable-line
+
+  const intents = [[NEO, neoToMint], [GAS, gasToMint]]
```

Comment:
I don't think `NEO` and `GAS` are symbols here, should these be `ASSETS.NEO` and `ASSETS.GAS`?